IHP/Controller/AccessDenied.hs

```diff
@@ -0,0 +1,158 @@
+module IHP.Controller.AccessDenied
+    ( accessDeniedWhen
+    , accessDeniedUnless
+    , handleAccessDeniedFound
+    , buildAccessDeniedResponse
+    )
+ where
+
+import IHP.Prelude hiding (displayException)
+import IHP.Controller.RequestContext
+import Network.HTTP.Types (status403)
+import Network.Wai
+import Network.HTTP.Types.Header
+import qualified Text.Blaze.Html.Renderer.Utf8 as Blaze
+import qualified Data.ByteString.Lazy as LBS
+import IHP.HSX.QQ (hsx)
+import qualified System.Directory as Directory
+import IHP.Controller.Context
+import IHP.Controller.Response (respondAndExit)
+
+
+-- | Stops the action execution with an access denied message (403) when the access condition is True.
+--
+-- __Example:__ Checking a user is the author of a blog post.
+--
+-- > action EditPostAction { postId } = do
+-- >     post <- fetch postId
+-- >     accessDeniedWhen (post.authorId /= currentUserId)
+-- >
+-- >     renderHtml EditView { .. }
+--
+-- This will throw an error and prevent the view from being rendered when the current user is not the author of the post.
+accessDeniedWhen :: (?context::ControllerContext) => Bool -> IO ()
+accessDeniedWhen condition = when condition renderAccessDenied
+
+-- | Stops the action execution with an access denied message (403) when the access condition is False.
+--
+-- __Example:__ Checking a user is the author of a blog post.
+--
+-- > action EditPostAction { postId } = do
+-- >     post <- fetch postId
+-- >     accessDeniedUnless (post.authorId == currentUserId)
+-- >
+-- >     renderHtml EditView { .. }
+--
+-- This will throw an error and prevent the view from being rendered when the current user is not the author of the post.
+accessDeniedUnless :: (?context::ControllerContext) => Bool -> IO ()
```

Comment:
```suggestion
accessDeniedUnless :: (?context :: ControllerContext) => Bool -> IO ()
```

Reply:
Wish we had an `Elm format` equivalent, that "simply works" without any settings

Reply:
yeah true :D